### PR TITLE
Bump Tentacle Version to add full agent upgrade support

### PR DESCRIPTION
**Pull request by scme0**
# Background

The Server and Tentacle changes have been merged to allow server to run raw scripts without an active NFS mount. This allows server to upgrade the agent even if the NFS pod needs to be restarted.

# Results

Bumping Tentacle to this version brings the functionality officially the the Kubernetes agent. I've also updated the test to use the `IsRawScript` option in the upgrade test to ensure that the changes are tested.

### CODE REVIEW

File: tests/kubernetes-agent/KubernetesAgent.IntegrationTests/Setup/Common/TestLogger.cs [hunk 1]
```diff
@@ -1,225 +1,37 @@
 using System;
-using Octopus.Diagnostics;
-using ILog = Octopus.Diagnostics.ILog;
+using Octopus.Tentacle.Contracts.Logging;
 
 namespace KubernetesAgent.Integration.Setup.Common
 {
-    public sealed class TestLogger(ILogger logger) : ILog, IDisposable
```
Comment (scme0) [inline tests/kubernetes-agent/KubernetesAgent.IntegrationTests/Setup/Common/TestLogger.cs:7]: `Tentacle.Client` no longer references `Octopus.Diagnostics` (which has been archived) so we have to use the new `Tentacle.Client` logger interface `ITentacleClientTaskLog`.

File: tests/kubernetes-agent/KubernetesAgent.IntegrationTests/KubernetesAgent.IntegrationTests.csproj [hunk 2]
```diff
@@ -13,8 +13,8 @@
     <ItemGroup>
         <PackageReference Include="coverlet.collector" Version="6.0.0"/>
         <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0"/>
-        <PackageReference Include="Octopus.Client" Version="14.3.1381" />
```
Comment (scme0) [inline tests/kubernetes-agent/KubernetesAgent.IntegrationTests/KubernetesAgent.IntegrationTests.csproj:16]: Upgraded `Octopus.Client` while I was in here.

File: .changeset/mighty-melons-sniff.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"kubernetes-agent": minor
+---
+
+Bump Tentacle version (to 8.1.1819) to add full server-driven upgrade support
```
Comment (APErebus) [inline .changeset/mighty-melons-sniff.md:5]:
```suggestion
Upgrade kubernetes-agent-tentacle to 8.1.1819 to support more reliable server-driven upgrades
```

**Review by APErebus — APPROVED** [review]
Excellent. Minor nit on versioning message, but :+1:

**Pull request merged**